p2p/p2p.go

```diff
@@ -188,3 +203,89 @@ func (s *Service) ListenAddrs() ([]multiaddr.Multiaddr, error) {
 
 	return listenAddrs, nil
 }
+
+// NewStream creates a bidirectional connection to a random peer that implements a set of protocol ids
+func (s *Service) NewStream(ctx context.Context, pids ...protocol.ID) (network.Stream, error) {
+	peers := s.host.Peerstore().Peers()
+	peersCount := peers.Len()
+	if peersCount < 0 {
+		return nil, errors.New("no peers")
+	}
+
+	randomPeerIdx := rand.Intn(peersCount) //nolint: gosec
```

Comment:
I haven't found `rand.Seek(...)` in our codebase so probably for the same number of peers `rand.Intn(...)` (it's pseudorandom) will return the same index over and over again. Could there be any issues with it?